pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java

```diff
@@ -256,13 +253,9 @@ public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
       if (_selections[i]) {
         TransformFunction transformFunction = _elseThenStatements.get(i);
         int[] intValues = transformFunction.transformToIntValuesSV(projectionBlock);
-        if (_numSelections == 1) {
```

Comment:
this optimisation is removed because it can only trigger if only the else statement matches, which is unlikely in the typical case.